ui/apps/platform/src/types/databaseService.proto.ts

```diff
@@ -0,0 +1,5 @@
+export type DatabaseStatus = {
+    databaseAvailable: boolean;
+    databaseType?: string | null;
+    databaseVersion?: string | null;
```

Comment:
@dashrews78 Can you help clarify a few points on alignment of frontend and backend service?
@vjwilson Some recommended changes pending the clarification.

1. Concerning `?` and `| null`
    * Is the following comment from backend code `// Only return the database type and version to logged in users, not anonymous users.` the explanation why? If yes, Van please copy it to the type definition.
    * Is `null` value possible? I have seen `null` value for some properties which have `string whatever` in proto file. I did not read the Go code and unit tests deeply enough to answer this question.
2. Is database status conceptually a metadata service, or is it coincidence left over from initial attempt?
    * If it is, then from my viewpoint as reviewer or future maintainer, better to move both frontend type and function into MetadataService.ts file to align with corresponding backend file.
    * If it is not, is it possible to move the backend service to a separate file?

Reply:
@pedrottimark The initial attempt was to simply use the metadata endpoint but that hosed up sensors so we had to pivot.  To me this is a metadata service of the likes of metadata.  The idea is databaseType and databaseVersion are useful information, but like the version in metadata endpoint we probably don't want to provide those details if unauthenticated.  I don't think you will receive a null unless your (java|type)script interprets an empty string as such.  Here is what a curl will provide you:

```
{"databaseAvailable":false,"databaseType":"Hidden","databaseVersion":""}
```

notice the databaseVersion is an empty string.

In regard to your 2nd point, in my mind it is conceptually a metadata service which is why I put it where I did on the backend.  Though I would disagree that the backend and UI need to house the related.  Things such as that could move at any time and keeping where the backend and UI house code seems like an unnecessary task.  IMO

Reply:
The majority of my contributions are incremental fixes and updates in files that are new to me.

Reasons that caused me to align the files when I rewrote JavaScript as TypeScript:
* I needed to compare the services and types. Where were they? Wasted time to search and find.
* When there is an incremental change to add a field to an existing response, Europe engineers are willing and able to accelerate progress by doing both. Where are they? Wasted time to search and find.